dispatch/tests/test_api_imagegallery.py

```diff
@@ -0,0 +1,380 @@
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        url = reverse('api-galleries-detail', args=[id])
+        response = self.client.get(url, format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        # Check that the data was updated
+        self.assertEqual(response.data['title'], 'New Gallery Title')
+        self.assertEqual(response.data['images'][0]['caption'], 'new test caption 1')
+        self.assertEqual(response.data['images'][0]['image']['id'], img_id2)
+        self.assertEqual(response.data['images'][1]['caption'], 'new test caption 2')
+        self.assertEqual(response.data['images'][1]['image']['id'], img_id1)
+
+    def test_imagegallery_patch_invalid_id(self):
+        """Ensure that patching invalid id returns the appropriate error"""
+
+        id = -1
+        url = reverse('api-galleries-detail', args=[id])
+        response = self.client.patch(url, format='json')
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_imagegallery_patch_invalid_image_ids(self):
+        """Ensure that an appropriate error is returned when an image gallery is
+        patched with an invalid image id"""
+
+        gallery, img_id1, img_id2 = self._create_gallery(1)
+        id = gallery.data['id']
+
+        url = reverse('api-galleries-detail', args=[id])
+
+        data = {
+            "title": "New Gallery Ttile",
+            "attachment_json": [
+                {
+                    "caption": "new test caption",
+                    "image_id": -1
+                }
+            ]
+        }
+
+        response = self.client.patch(url, data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_imagegallery_patch_no_image_ids(self):
+        """Ensure that an appropriate error is returned when an image gallery is
+        patched with an invalid image id"""
+
+        gallery, img_id1, img_id2 = self._create_gallery(1)
+        id = gallery.data['id']
+
+        url = reverse('api-galleries-detail', args=[id])
+
+        data = {
+            "title": "New Gallery Title",
+            "attachment_json": [
+                {
+                    "caption": "new test caption"
+                }
+            ]
+        }
+
+        response = self.client.patch(url, data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_imagegallery_delete_unauthorized(self):
+        """Delete gallery should fail with unauthenticated request"""
+
+        gallery, img_id1, img_id2 = self._create_gallery(1)
+        id = gallery.data['id']
+
+        # Clear authentication credentials
+        self.client.credentials()
+
+        url = reverse('api-galleries-detail', args=[id])
+        response = self.client.delete(url, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
```

Comment:
I forgot to mention this earlier, but we should check to make sure the gallery wasn't deleted after this call is made. This can be done either by making another API GET request or by querying the database manually like so:

```py
try:
  ImageGallery.objects.get(pk=id)
except ImageGallery.DoesNotExist:
  self.fail('The gallery should not have been deleted')
```